src/libsyntax/parse/parser.rs

```diff
@@ -767,6 +767,9 @@ impl<'a> Parser<'a> {
             err.span_label(self.span, format!("expected identifier, found {}", token_descr));
         } else {
             err.span_label(self.span, "expected identifier");
+            if self.token == token::Comma {
+                err.span_suggestion(self.span, "remove this comma", ",".into());
```

Comment:
The third argument here [should be an empty string](https://github.com/rust-lang/rust/pull/51184/files#diff-da9d34ca1d0f6beee2838cf02e07345cR2515). This will change the output to `help: remove this comma`.